client/my-sites/themes/themes-banner/upwork.jsx

```diff
@@ -0,0 +1,123 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import Button from 'components/button';
+import safeImageUrl from 'lib/safe-image-url';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { recordTracksEvent as recordTracksEventAction } from 'state/analytics/actions';
+import { isThemesUpworkBannerVisible } from 'state/themes/themes-ui/selectors';
+import { hideThemesUpworkBanner } from 'state/themes/themes-ui/actions';
+
+class UpworkBanner extends PureComponent {
+	static propTypes = {
+		// Connected props
+		siteId: PropTypes.number,
+		currentPlan: PropTypes.object,
+		isBannerVisible: PropTypes.bool,
+		hideThemesBanner: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.recordView();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.siteId && this.props.siteId && this.props.siteId !== prevProps.siteId ) {
+			this.recordView();
+		}
+	}
+
+	recordView() {
+		if ( this.props.isBannerVisible ) {
+			this.props.recordTracksEvent( 'calypso_upwork_themes_banner_view' );
+		}
+	}
+
+	recordClick = eventName => {
+		const plan = this.props.plan ? this.props.plan.productSlug : '';
```

Comment:
looks like this should be `this.props.currentPlan`. I am only getting blank strings in the tracks events
```suggestion
		const plan = this.props.currentPlan ? this.props.currentPlan.productSlug : '';
```